engine/src/cgimageutil.cpp

```diff
@@ -104,18 +114,14 @@ bool MCGRasterToCGImage(const MCGRaster &p_raster, MCGRectangle p_src_rect, CGCo
 	t_width = p_src_rect.size.width;
 	t_height = p_src_rect.size.height;
 	
-	/* OVERHAUL - REVISIT: pixel formats */
-	const uint8_t *t_src_ptr = (uint8_t*)p_raster.pixels;
-	t_src_ptr += t_y * p_raster.stride + t_x * sizeof(uint32_t);
+	const uint8_t *t_src_ptr = (uint8_t*)MCGRasterGetPixelPtr(p_raster, t_x, t_y);
```

Comment:
As stylistic improvement, do you think it might be a good idea to change `MCGRasterGetPixelPtr()` to return `uint8_t *` or `byte_t *`? It would still be possible to store the result into a `void *` but maybe it would be a more informative prototype.